src/server/rdb_save.cc

```diff
@@ -337,13 +337,6 @@ error_code RdbSerializer::SelectDb(uint32_t dbid) {
 // Called by snapshot
 io::Result<uint8_t> RdbSerializer::SaveEntry(const PrimeKey& pk, const PrimeValue& pv,
                                              uint64_t expire_ms, uint32_t mc_flags, DbIndex dbid) {
-  if (!pv.TagAllowsEmptyValue() && pv.Size() == 0) {
```

Comment:
IMO this doesn't make much sense since the value will be deleted when an `hget` variants gets called and the underline dense set becomes empty. I guess you don't want anything more from here @romange

Reply:
You assume that the bug is fixed. I can tell you it's not fully fixed, so I prefer to keep it. Moreover, I would like to convert `LOG(ERROR)` below to `LOG(DFATAL)`

Reply:
Why do you think I assume that the bug is fixed? We haven't fixed anything IMO we are just dealing with the issue. The bug will be fixed once we don't rely in "touching" the dense set to remove it. In other words:

```
  EXPECT_THAT(Run({"HSET", "foo", "a_field", "a_value"}), IntArg(1));
  EXPECT_THAT(Run({"HSETEX", "foo", "1", "b_field", "b_value"}), IntArg(1));
  EXPECT_THAT(Run({"HDEL", "foo", "a_field"}), IntArg(1));
  //advance time here
   EXPECT_THAT(Run({"dbsize"}), IntArg(0));
```

+1 for `DFATAL` it would be nice to be aware of this in debug builds and actually cause a failure